### PR TITLE
encryption: retry requests during e2e when kube-apiserver restarts

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: test/library/encryption/helpers.go [hunk 1]
```diff
@@ -227,7 +233,7 @@ func ForceKeyRotation(t testing.TB, updateUnsupportedConfig UpdateUnsupportedCon
 		return err
 	}
 
-	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+	return onErrorWithTimeout(wait.ForeverTestTimeout, retry.DefaultBackoff, orError(errors.IsConflict, transientAPIError), func() error {
```
Comment (p0lyn0mial) [inline test/library/encryption/helpers.go]: do we need `onError` since `IsConflict` is already considered `transientError` ?

**Comment by p0lyn0mial** [comment]
potentially we have many places that would like to use `onErrorWithTimeout` function not only `GetLastKeyMeta`.

File: test/library/encryption/helpers.go (hunk 1 above)
Comment (sttts) [reply test/library/encryption/helpers.go]: IsConflict is not transient

**Pull request merged**